### PR TITLE
[Feature] support mutable config for star cache

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 4 / 10 (40.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/http/action/update_config_action.cpp|0|3|00.00%|[143, 144, 145]|
|:large_blue_circle:|src/service/staros_worker.cpp|4|7|57.14%|[387, 388, 389]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/http/action/update_config_action.cpp|0|1|00.00%|[144]|
|:large_blue_circle:|src/service/staros_worker.cpp|0|4|00.00%|[383, 384, 385, 386]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/http/action/update_config_action.cpp|0|1|00.00%|[144]|
|:large_blue_circle:|src/service/staros_worker.cpp|0|4|00.00%|[383, 384, 385, 386]|

**Review by kevincai — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/http/action/update_config_action.cpp|0|1|00.00%|[144]|
|:large_blue_circle:|src/service/staros_worker.cpp|0|4|00.00%|[383, 384, 385, 386]|

File: be/src/service/staros_worker.cpp [hunk 1]
```diff
@@ -378,5 +379,13 @@ void shutdown_staros_worker() {
     staros::starlet::common::ShutdownHook::shutdown();
 }
 
+// must keep each config the same
+void update_staros_starcache() {
+    if (fslib::FLAGS_use_star_cache != config::starlet_use_star_cache) {
+        fslib::FLAGS_use_star_cache = config::starlet_use_star_cache;
+        (void)fslib::star_cache_init(fslib::FLAGS_use_star_cache);
```
Comment (wyb) [inline be/src/service/staros_worker.cpp:386]: what if `starlet_use_star_cache` is updated from `true` to `false`

Comment (starrocks-xupeng) [reply be/src/service/staros_worker.cpp:386]: it's ok, it will be handled correctly in `fslib::star_cache_init`

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by starrocks-xupeng** [comment]
@mergify backport branch-3.2 branch-3.1

**Pull request merged**